projects/core/directives/hint/hint.component.ts

```diff
@@ -108,8 +109,9 @@ export class TuiHintComponent<C = any> {
     @HostListener('document:click', ['$event.target'])
     onClick(target: HTMLElement): void {
         if (
-            !this.el.nativeElement.contains(target) &&
-            !this.hover.el.nativeElement.contains(target)
+            (!this.el.nativeElement.contains(target) &&
+                !this.hover.el.nativeElement.contains(target)) ||
+            tuiGetElementObscures(this.hover.el.nativeElement)?.length
```

Comment:
Can you describe this scenario please?

Reply:
yes, this PR solves two scenarios (you can check it [here](https://taiga-ui.dev/components/dialog#directive)):

1. If hint already displayed when we click the button. `click` event bubbles to document => hint is obscured by dialog => close hint (most common case for desktop safari). This code is for this case

2. On mobile, hint doesn't appear until we tap the button, `document.click` handler doesn't work for this case. So we have to check after "hover" if is hint obscured.  `map(value => value && !tuiGetElementObscures(this.el.nativeElement)?.length)` is for this case.